crates/sui-types/src/programmable_transaction_builder.rs

```diff
@@ -14,27 +14,32 @@ use crate::{
     messages::{
         Argument, CallArg, Command, MoveCall, MoveModulePublish, ObjectArg, Pay, PayAllSui, PaySui,
         ProgrammableMoveCall, ProgrammableTransaction, SingleTransactionKind, TransactionData,
-        TransactionDataAPI, TransactionKind, TransferObject, TransferSui,
+        TransactionKind, TransferObject, TransferSui,
     },
 };
 
-pub fn migrate_transaction_data(mut m: TransactionData) -> anyhow::Result<TransactionData> {
+pub fn migrate_transaction_data(m: TransactionData) -> anyhow::Result<TransactionData> {
     let mut builder = ProgrammableTransactionBuilder::new();
-    match m.kind().clone() {
+    let TransactionData::V1(mut v1) = m;
+    match v1.kind {
         TransactionKind::Single(SingleTransactionKind::PaySui(PaySui {
```

Comment:
Do we plan to just remove these existing `SingleTransactionKind` before March 15th?

Reply:
yes